### DISCUSSION TITLE
How to translate the Carpet Mod and extensions (mods)?

### DISCUSSION THREAD

**Discussion by Amirhan-Taipovjan-Greatest-I** [gnembon/fabric-carpet#1737] [Q&A]
Hello!

I actually don't know how to translate all content of the Carpet Mod... I checked translation files, but I still can't find command output strings and etc. Is it possible to translate the Carpet Mod fully?

**Comment by altrisi** [comment]
Not everything, currently it's a bit limited and only the `/carpet` command and the rule information can be translated, and the language has to be set server-side (via `/carpet language`).

To translate what's currently possible you have to:
- Add the language id to this line: https://github.com/gnembon/fabric-carpet/blob/ad4a41e4973cfb98723870d2b9801c5be18e454f/src/main/java/carpet/CarpetSettings.java#L74
- Create a json file called what the id is in `src/main/resources/assets/carpet/lang/` and fill it with the language:
  - The base English file with everything translatable is currently here: https://github.com/altrisi/fabric-carpet/blob/migrate-rules/src/main/resources/assets/carpet/lang/en_us.json
  - You can also translate the rule names with `carpet.rule.[ruleId].name`. The english id will still be show next to the name (given it's what's used in the command). For example:
  `"carpet.rule.chainStone.name": "CadenaPiedra"`
